include/podio/UserDataCollection.h

```diff
@@ -0,0 +1,229 @@
+#ifndef USERDATACOLLECTION_H
+#define USERDATACOLLECTION_H
+
+#include "podio/CollectionBase.h"
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+#include <typeindex>
+
+#define PODIO_ADD_USER_TYPE(type) template<> constexpr const   char* userDataTypeName<type>(){ return #type ; }
+
+
+namespace podio {
+
+
+
+
+  namespace detail {
+
+    // some templates to ensure valid and supported user types
+    // as suggested by T.Madlener, DESY
+
+    /** tuple of basic types supported in user vector
+     */
+    using SupportedUserDataTypes = std::tuple<
+      int, long, float, double,
+      unsigned int, unsigned, unsigned long,
+      char, short, long long,
```

Comment:
can you quickly clean up the non-fixed width types?

Reply:
I actually intentionally left them in, as this is what we currently support in PODIO. I think we should address the issue of 'set of supported types' in PODIO consistently in a dedicate PR (with sufficient documentation). What do you think?

Reply:
I think for a new feature we should already start restricted. We can open up easily later in case. The other way round is more complicated.